src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,568 @@
+            gdf_adm = gdf[gdf[feature_col] == feature]
+
+            da_clip = data_obj.rio.set_spatial_dims(
+                x_dim=self.x_dim, y_dim=self.y_dim
+            )
+
+            # clip returns error if no overlapping raster cells for geometry
+            # so catching and skipping rest of iteration so no stats computed
+            try:
+                da_clip = da_clip.rio.clip(
+                    gdf_adm[geom_col], all_touched=all_touched
+                )
+            except NoDataInBounds:
+                logger.warning(
+                    "No overlapping raster cells for %s, skipping.", feature
+                )
+                continue
+
+            grid_stat_all = []
+            for stat in stats_list:
+                # count automatically ignores NaNs
+                # therefore skipna can also not be given as an argument
+                # implemented count cause needed for computing percentages
+                kwargs: Dict[str, Any] = {}
+                if stat != "count":
+                    kwargs["skipna"] = True
+                # makes sum return NaN instead of 0 if array
+                # only contains NaNs
+                if stat == "sum":
+                    kwargs["min_count"] = 1
+                grid_stat = getattr(da_clip, stat)(
+                    dim=[self.x_dim, self.y_dim], **kwargs
+                ).rename(f"{stat}_{feature_col}")
+                grid_stat_all.append(grid_stat)
+
+            if percentile_list is not None:
+                grid_quant = [
+                    da_clip.quantile(quant / 100, dim=[self.x_dim, self.y_dim])
+                    .drop("quantile")
+                    .rename(f"{quant}quant_{feature_col}")
+                    for quant in percentile_list
+                ]
+                grid_stat_all.extend(grid_quant)
+
+            # if dims is 0, it throws an error when merging
+            # and then converting to a df
+            # this occurs when the input da is 2D
+            if not grid_stat_all[0].dims:
+                df_adm = pd.DataFrame(
+                    {
+                        da_stat.name: [da_stat.values]
+                        for da_stat in grid_stat_all
+                    }
+                )
+
+            else:
+                zonal_stats_xr = xr.merge(grid_stat_all)
+                df_adm = (
+                    zonal_stats_xr.to_dataframe()
+                    .drop("spatial_ref", axis=1)
+                    .reset_index()
+                )
+            df_adm[feature_col] = feature
+            df_list.append(df_adm)
+
+        df_zonal_stats = pd.concat(df_list).reset_index(drop=True)
+        return df_zonal_stats
+
+
+@xr.register_dataset_accessor("aat")
+class AatRasterDataset(AatRasterMixin, RasterDataset):
+    """AA toolbox extension for xarray.Dataset."""
+
+    def __init__(self, xarray_object):
+        super().__init__(xarray_object)
+
+    def get_raster_array(self, var: str) -> xr.DataArray:
```

Comment:
Annoying that you have to write this method, but good catch. Does `rioxarray` try to handle this at all?

Reply:
From looking through [their code](https://github.com/corteva/rioxarray/blob/master/rioxarray/raster_dataset.py), I didn't see any attempt to deal with it.